File: 2015/04_the_ideal_stocking_stuffer/part1.py
import hashlib

i = 0
while True:
    i += 1

    text = "yzbqklnj" + str(i)

    m = hashlib.md5()
    m.update(text.encode('UTF-8'))
    if m.hexdigest().startswith("00000"):
        print(i)
        break

File: 2015/04_the_ideal_stocking_stuffer/part1.go
package main

import (
	"crypto/md5"
	"encoding/hex"
	"fmt"
	"strconv"
	"strings"
)

func getMD5Hash(text string) string {
	hash := md5.Sum([]byte(text))
	return hex.EncodeToString(hash[:])
}

func main() {
	i := 0
	for true {
		i += 1
		s := strconv.Itoa(i)

		text := "yzbqklnj" + s

		hash := getMD5Hash(text)
		if strings.HasPrefix(hash, "00000") {
			fmt.Println(i)
			break
		}
	}
}
